runtime/reconcilers/util.go

```diff
@@ -99,17 +100,39 @@ func olapDropTableIfExists(ctx context.Context, c *runtime.Controller, connector
 	}
 	defer release()
 
-	var typ string
-	if view {
-		typ = "VIEW"
-	} else {
-		typ = "TABLE"
-	}
+	_ = olap.WithConnection(ctx, 100, false, false, func(ctx, ensuredCtx context.Context, conn *sql.Conn) error {
+		if view {
+			_, err = conn.ExecContext(ctx, fmt.Sprintf("DROP VIEW IF EXISTS %s", safeSQLName(table)))
+			if err != nil {
+				return err
+			}
+			rows, err := conn.QueryContext(ensuredCtx, "SELECT table_name FROM information_schema.tables WHERE table_schema='rill_sources'")
+			if err != nil {
+				return err
+			}
 
-	_ = olap.Exec(ctx, &drivers.Statement{
-		Query:       fmt.Sprintf("DROP %s IF EXISTS %s", typ, safeSQLName(table)),
-		Priority:    100,
-		LongRunning: true,
+			var tableName string
+			var lastErr error
+			srcRegex, err := regexp.Compile(fmt.Sprintf("__%s_[0-9]+$", table))
+			if err != nil {
+				return err
+			}
+			for rows.Next() {
+				if err := rows.Scan(&tableName); err != nil {
+					break
+				}
+				if !srcRegex.MatchString(tableName) {
+					continue
+				}
+				if _, err = conn.ExecContext(ensuredCtx, fmt.Sprintf("DROP TABLE IF EXISTS rill_sources.%s", safeSQLName(tableName))); err != nil {
+					lastErr = err
+				}
+			}
+			return lastErr
+		}
+		// table
+		_, err = conn.ExecContext(ctx, fmt.Sprintf("DROP TABLE IF EXISTS %s", safeSQLName(table)))
+		return err
 	})
```

Comment:
1. Needs to pass `true, false` to `WithConnection` (not `false, false`)
2. Reduce nesting by handling the table case early and exiting (see [reduce nesting](https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting) in the style guide)
3. Also, you can't use the `conn` to execute a new query until the `rows` from the previous case has been closed. Dropping the shadow table should happen after the find step, not inside the loop.
4. Regex compiles are kind of expensive, and I'm a bit worried about edge cases of `sprintf` + regex compile. Wouldn't it be sufficient to do a prefix check? I would also recommend doing a double underscore `__` between the base name and numbers.

Reply:
1. I expected these queries to run fast so that's why marked long running as false. Is it always expected to be true for `WithConnection` ?
4. Without regex there can be some edge cases if user has two sources named `test` and `test__1` so with a prefix check underlying tables of both will pass the check.

Reply:
May be we can add some other string between table name and timestamp but that will still not be 100% reliable.

Reply:
1. Dropping a table won't necessarily run very fast, so best to mark it long-running
2. I don't think 100% reliable is possible in any case. But yeah something in between would increase safety, something like `[name]__shadow[ID]` should be pretty safe